com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/processes/metering/MicrometerNotifier.java

```diff
@@ -0,0 +1,136 @@
+package com.sap.cloud.lm.sl.cf.processes.metering;
+
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.DEFAULT_TAG_VALUE;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.END_PROCESS_EVENT_MULTIAPPS_METRIC;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.ERROR_PROCESS_EVENT_MULTIAPPS_METRIC;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.OVERALL_PROCESS_TIMER_MULTIAPPS_METRIC;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.START_PROCESS_EVENT_MULTIAPPS_METRIC;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.STEP_PROCESS_TIMER_MULTIAPPS_METRIC;
+import static com.sap.cloud.lm.sl.cf.processes.metering.MicrometerConstants.*;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.sap.cloud.lm.sl.cf.process.Messages;
+import com.sap.cloud.lm.sl.cf.process.steps.StepPhase;
+import com.sap.cloud.lm.sl.cf.process.util.ProcessTypeParser;
+import com.sap.cloud.lm.sl.cf.process.variables.VariableHandling;
+import com.sap.cloud.lm.sl.cf.process.variables.Variables;
+import com.sap.cloud.lm.sl.cf.web.api.model.Operation.State;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.dynatrace.DynatraceMeterRegistry;
+
+@Named
+public class MicrometerNotifier {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MicrometerNotifier.class);
+
+    @Inject
+    private ProcessTypeParser processTypeParser;
+    private DynatraceMeterRegistry dynatraceMeterRegistry;
+
+    public MicrometerNotifier(@Autowired(required = false) DynatraceMeterRegistry dynatraceMeterRegistry) {
```

Comment:
If you're using constructor-injection, inject the `processTypeParser` in the constructor as well. Not a good idea to mix constructor with field injection.